File: iteration2/src/models/CourseSection.java
package iteration2.src.models;

import iteration2.src.utils.DatabaseManager;
import iteration2.src.utils.Util;

import java.util.List;
import java.util.ArrayList;

public class CourseSection {

    private int studentCapacity;
    private String lecturerName;
    private ArrayList<String> sectionTime; // [8:00, 9:00, 8:00]
    private ArrayList<String> sectionDay; // [Monday, Monday, Tuesday]
    private String classroom;
    private String sectionCode;
    private int studentCountInsideCourseSection;

    public CourseSection() {
    }

    public CourseSection(int studentCapacity, String lecturerName, ArrayList<String> sectionTime,
            ArrayList<String> sectionDay, String classroom, String sectionCode) {
        this.studentCapacity = studentCapacity;
        this.lecturerName = lecturerName;
        this.sectionTime = sectionTime;
        this.sectionDay = sectionDay;
        this.classroom = classroom;
        this.sectionCode = sectionCode;
        this.studentCountInsideCourseSection = 0;
    }

    public boolean checkAvailibilty() {
        if (studentCapacity >= studentCountInsideCourseSection) {
            return true;
        } else {
            return false;
        }
    }

    public int incrementStudentCount() {
        studentCountInsideCourseSection++;
        Util.getLogger().info("Student count incremented. Current section count: " + studentCountInsideCourseSection);
        return studentCountInsideCourseSection;
    }

    public int decrementStudentCount() {
        studentCountInsideCourseSection--;
        Util.getLogger().info("Student count decremented. Current section count: " + studentCountInsideCourseSection);
        return studentCountInsideCourseSection;
    }

    public Course findCourseOfCourseSection() {
        List<Course> courses = DatabaseManager.getInstance().getCourses();
        for (Course course : courses) {

            String courseCode = this.getSectionCode().substring(0, this.getSectionCode().indexOf("."));
            if (course.getCourseCode().equals(courseCode)) {
                return course;
            }
        }
        return null;
    }

    public int getStudentCapacity() {
        return studentCapacity;
    }

    public String getLecturerName() {
        return lecturerName;
    }

    public ArrayList<String> getSectionTime() {
        return sectionTime;
    }

    public String getClassroom() {
        return classroom;
    }

    public String getSectionCode() {
        return sectionCode;
    }

    public int getStudentCountInsideCourseSection() {
        return studentCountInsideCourseSection;
    }

    public ArrayList<String> getSectionDay() {
        return sectionDay;
    }

    public void setStudentCapacity(int studentCapacity) {
        this.studentCapacity = studentCapacity;
    }

    public void setLecturerName(String lecturerName) {
        this.lecturerName = lecturerName;
    }

    public void setSectionTime(ArrayList<String> sectionTime) {
        this.sectionTime = sectionTime;
    }

    public void setSectionDay(ArrayList<String> sectionDay) {
        this.sectionDay = sectionDay;
    }

    public void setClassroom(String classroom) {
        this.classroom = classroom;
    }

    public void setSectionCode(String sectionCode) {
        this.sectionCode = sectionCode;
    }

    public void setStudentCountInsideCourseSection(int studentCountInsideCourseSection) {
        this.studentCountInsideCourseSection = studentCountInsideCourseSection;
    }

}

File: iteration3/src/models/CourseSection.py
import logging
from iteration3.src.utils.DatabaseManager import DatabaseManager

class CourseSection:

    def __init__(self, studentCapacity, lecturerName, sectionTime, sectionDay, classroom, sectionCode, studentCountInsideCourseSection):
        self.studentCapacity = studentCapacity
        self.lecturerName = lecturerName
        self.sectionTime = sectionTime
        self.sectionDay = sectionDay
        self.classroom = classroom
        self.sectionCode = sectionCode
        self.studentCountInsideCourseSection = studentCountInsideCourseSection

    @classmethod
    def dictToObject(cls, dict):
        return cls(
            dict['studentCapacity'],
            dict['lecturerName'],
            dict['sectionTime'],
            dict['sectionDay'],
            dict['classroom'],
            dict['sectionCode'],
            dict['studentCountInsideCourseSection']
        )

    def checkAvailability(self):
        if self.studentCapacity >= self.studentCountInsideCourseSection:
            return True
        else:
            return False
        
    def incrementStudentCount(self):
        self.studentCountInsideCourseSection += 1
        logging.log(logging.INFO, "Student count incremented. Current section count: " + str(self.studentCountInsideCourseSection))
        return self.studentCountInsideCourseSection
    
    def decrementStudentCount(self):
        self.studentCountInsideCourseSection -= 1
        logging.log(logging.INFO, "Student count decremented. Current section count: " + str(self.studentCountInsideCourseSection))
        return self.studentCountInsideCourseSection
    
    def findCourseOfCourseSection(self):
        index = self.sectionCode.index(".")
        courseCode = self.sectionCode[:index]

        courses = DatabaseManager().getInstance().getCourseList()
        for course in courses:
            if course.getCourseCode() == courseCode:
                return course
        return None
    
    def getSectionCode(self):
            return self.sectionCode
        
    def getStudentCapacity(self):
        return self.studentCapacity
        
    def getLecturerName(self):
        return self.lecturerName
        
    def getSectionTime(self):
        return self.sectionTime
        
    def getSectionDay(self):
        return self.sectionDay
        
    def getClassroom(self):
        return self.classroom
        
    def getStudentCountInsideCourseSection(self):
        return self.studentCountInsideCourseSection        
            
    
    
    
        






        